src/app/js/formats/vega-lite/models/BarChart.js

```diff
@@ -187,26 +174,10 @@ class BarChart extends BasicChart {
         model.encoding.x.axis.labelAngle = this.labelAngle.x;
         model.encoding.y.axis.labelAngle = this.labelAngle.y;
 
-        switch (this.orderBy) {
-            case VALUES_ASC:
-                labelsModel.encoding.y.sort = model.encoding.x.sort =
-                    this.direction === AXIS_VERTICAL ? 'y' : 'x';
-                break;
-            case VALUES_DESC:
-                labelsModel.encoding.y.sort = model.encoding.x.sort =
-                    this.direction === AXIS_VERTICAL ? '-y' : '-x';
-                break;
-            case LABEL_ASC:
-                labelsModel.encoding.y.sort = model.encoding.x.sort =
-                    this.direction === AXIS_VERTICAL ? 'x' : 'y';
-                break;
-            case LABEL_DESC:
-                labelsModel.encoding.y.sort = model.encoding.x.sort =
-                    this.direction === AXIS_VERTICAL ? '-x' : '-y';
-                break;
-            default:
-                break;
-        }
+        labelsModel.encoding.x.sort = null;
```

Comment:
Add a comment to say that you disable vega's sorts, because they're already done by routines (and because, when both are enabled, the sort was wrong, inexplicably).